app/controllers/predicates/InflightPPOBPredicate.scala

```diff
@@ -20,66 +20,62 @@ import javax.inject.{Inject, Singleton}
 
 import common.SessionKeys.inflightPPOBKey
 import config.{AppConfig, ErrorHandler}
-import controllers.BaseController
 import models.User
 import play.api.Logger
-import play.api.i18n.MessagesApi
-import play.api.mvc.{ActionBuilder, ActionFunction, Request, Result}
-import services.{EnrolmentsAuthService, VatSubscriptionService}
-import uk.gov.hmrc.auth.core.retrieve._
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{ActionRefiner, Result}
+import play.api.mvc.Results.{Ok, Redirect}
+import services.VatSubscriptionService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class InflightPPOBPredicate @Inject()(vatSubscriptionService: VatSubscriptionService,
-                                      enrolmentsAuthService: EnrolmentsAuthService,
                                       val errorHandler: ErrorHandler,
                                       val messagesApi: MessagesApi,
                                       implicit val appConfig: AppConfig,
-                                      implicit val ec: ExecutionContext) extends BaseController
-  with AuthBasePredicate with ActionBuilder[User] with ActionFunction[Request, User] {
+                                      implicit val ec: ExecutionContext)
+  extends ActionRefiner[User, User] with I18nSupport {
 
-  override def invokeBlock[A](request: Request[A], block: User[A] => Future[Result]): Future[Result] = {
+  override def refine[A](request: User[A]): Future[Either[Result, User[A]]] = {
 
-    implicit val req: Request[A] = request
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+    implicit val req: User[A] = request
```

Comment:
You could call this `user` rather that `req` just so that you know that the request is a WrappedRequest for the user - but that's just semantics.